src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.cs

```diff
@@ -323,6 +328,16 @@ private void DeleteAllWorkloadInstallationRecords()
 
         public IWorkloadInstallationRecordRepository GetWorkloadInstallationRecordRepository() => RecordRepository;
 
+        public void RemoveWorkloadManifest(string featureBand, string manifestId, string version, DirectoryPath? offlineCache)
```

Comment:
This may not be this simple. Since you're dealing with MSIs, you will need to verify that the dependencies allow you to remove the installation, or decide that if the current SDK band is the final dependent, that you can remove the dependent first and then uninstall the MSI.

Reply:
Great point! And fortunately, this is also dead code—I talked with dsplaisted about that a while ago, and we agreed to modify the install state then run garbage collection instead, which should resolve that issue, but then I forgot to remove this. Will fix 🙂

Reply:
As in the install state file? That doesn't change the dependents against the installer?

Reply:
What does that mean?

Reply:
The same MSIs can have multiple parents holding references to them. For workloads, that's usually the SDK. The dependents are stored in registry keys under `HKLM\SOFTWARE\Classes\Installer\Dependencies`. When an SDK releases its reference, the key is simply deleted. When there are no more dependents registered, the CLI can then remove the installation - this happens as part of the workload garbage collection.

Reply:
To be clear, you're saying that the garbage collector will not update that reference properly, so we should (manually?) delete references to the manifests we don't want in the registry?